src/main/java/seedu/address/newstorage/StorageManager.java

```diff
@@ -0,0 +1,77 @@
+package seedu.address.newstorage;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.logging.Logger;
+
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.commons.exceptions.DataConversionException;
+import seedu.address.model.ReadOnlyRecipeBook;
+import seedu.address.model.ReadOnlyUserPrefs;
+import seedu.address.model.UserPrefs;
+/**
+ * Manages storage of AddressBook data in local storage.
+ */
+public class StorageManager implements Storage {
+
+    private static final Logger logger = LogsCenter.getLogger(StorageManager.class);
+    private RecipeBookStorage recipeBookStorage;
+    private UserPrefsStorage userPrefsStorage;
+
+    /**
+     * Creates a {@code StorageManager} with the given {@code recipeBookStorage} and {@code UserPrefStorage}.
+     */
+    public StorageManager(RecipeBookStorage recipeBookStorage, UserPrefsStorage userPrefsStorage) {
+        this.recipeBookStorage = recipeBookStorage;
+        this.userPrefsStorage = userPrefsStorage;
+    }
+
+    // ================ UserPrefs methods ==============================
+
+    @Override
+    public Path getUserPrefsFilePath() {
+        return userPrefsStorage.getUserPrefsFilePath();
+    }
+
+    @Override
+    public Optional<UserPrefs> readUserPrefs() throws DataConversionException, IOException {
+        return userPrefsStorage.readUserPrefs();
+    }
+
+    @Override
+    public void saveUserPrefs(ReadOnlyUserPrefs userPrefs) throws IOException {
+        userPrefsStorage.saveUserPrefs(userPrefs);
+    }
+
+
+    // ================ AddressBook methods ==============================
+
+    @Override
+    public Path getRecipeBookFilePath() {
+        return recipeBookStorage.getRecipeBookFilePath();
+    }
+
+    @Override
+    public Optional<ReadOnlyRecipeBook> readRecipeBook() throws DataConversionException, IOException {
+        return readRecipeBook(recipeBookStorage.getRecipeBookFilePath());
+    }
+
+    @Override
+    public Optional<ReadOnlyRecipeBook> readRecipeBook(Path filePath) throws DataConversionException, IOException {
+        logger.fine("Attempting to read data from file: " + filePath);
+        return recipeBookStorage.readRecipeBook(filePath);
+    }
+
+    @Override
+    public void saveRecipeBook(ReadOnlyRecipeBook addressBook) throws IOException {
+        saveRecipeBook(addressBook, recipeBookStorage.getRecipeBookFilePath());
+    }
```

Comment:
minor: update the argument variable in this file:

```suggestion
    @Override
    public void saveRecipeBook(ReadOnlyRecipeBook recipeBook) throws IOException {
        saveRecipeBook(recipeBook, recipeBookStorage.getRecipeBookFilePath());
    }
```